File: leetcode/twoSum_test.go
package leetcode

import (
	"encoding/json"
	"fmt"
	"os"
	"reflect"
	"testing"
)

type TwoSumCast struct {
	Input struct {
		Nums   []int
		Target int
	}

	Output []int
}

func TestHelloName(t *testing.T) {
	content, _ := os.ReadFile("./cases.json")

	var payload []TwoSumCast
	_ = json.Unmarshal(content, &payload)

	for i, testCase := range payload {
		fmt.Printf("正在执行第 %d 个测试用例\n", i+1)
		fmt.Printf("nums: %v , target: %v ==> %v \n", testCase.Input.Nums, testCase.Input.Target, testCase.Output)

		result := twoSum()
		if !reflect.DeepEqual(result, testCase.Output) {
			t.Errorf("期望 %v, 结果 %v", testCase.Output, result)
		}
	}
}

File: leetcode/twoSum.test.ts
import cases from "./cases.json";
import twoSum from "./twoSum";

test("two sum", () => {
  cases.forEach((testCase, index) => {
    console.log(`正在执行第 ${index + 1} 个测试用例\n`);
    expect(twoSum(testCase.input.nums, testCase.input.target)).toStrictEqual(
      testCase.output
    );
  });
});
